code-challenges.test.js

```diff
@@ -16,35 +20,117 @@
+                                         //Tests:       1 failed, 1 total
 
 
 // b) Create the function that makes the test pass.
+        ///// INPUT: colors1, colors1
+        /////OUTPUT:["blue", "green", "yellow", "pink"] ; ["indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+
+                                ///Setting up the Jest testing framework, I discover a jest method call anything that will leave
+                        /// it relatively open to any test results. This is is help when I use the slice method to take off the 
+                        // first element in each array. I get a good failing test with my variable |shuffle|. Creating my function I    
+                        // plan to return the arrays by slicing the first element in each of the color1, color2 arrays. The problem is 
+                        // that the terminal output is [Function: shuffle]. it knows it's a function but for some reason my code is not running.
+                        // My function may not be invoked?
+
+         const shuffle = (arr) => {
+            return arr(colors1.slice(1))(colors2.slice(1))
+         }
+    
+
+ 
+
+
+
 
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = {upVotes: 13, downVotes: 2}
-// Expected output: 11
-const votes2 = {upVotes: 2, downVotes: 33}
-// Expected output: -31
+describe("tallyVote",() => {
+    it("returns the end tally that contains up votes and down votes.",() => {
+      const votes1 = {upVotes: 13, downVotes: 2}      // Expected output: 11
+    expect(tallyVote()).toEqual(11)
+
+         const votes2 = {upVotes: 2, downVotes: 33}     // Expected output: -31
+    expect(tallyVote()).toEqual(-31)
+      })
+})  
+//                         //  *****TEST RESULTS: Failed -----------> ReferenceError: tallyVote is not defined
+//                                          //Test Suites: 1 failed, 1 total
+//                                          //Tests:       1 failed, 1 total
 
 
 // b) Create the function that makes the test pass.
 
+                /////INPUT: votes1, votes2
+                ////OUTPUT: 11 ; -31
+
+                                ////I'm using tallyVote for my created function. The plan is to subtract each key 
+                         // value pair within each object, to the other key value pair to end up with the passing test. I begin
+                         // by returning each object |votes1 & votes2| subracting the key value pairs within. These are represented 
+                         // with "upVotes / downVotes". The terminal is stating that tallyVote is a function but it is not being executed.      
+       
+                         const tallyVote = (object) => {
+               return object (votes1.upVotes - votes1.downVotes)(votes2.upVotes - votes2.downVotes)
```

Comment:
Your logic looks good on how to approach this problem.  Since you have a parameter of object, this is allowing you to have dynamic code and never use a specific variable elsewhere. With that in mind, if you are passing votes1 or votes2 into your function in your test, you should be able to call on object.upVotes or object.downVotes to access those values.